### PR TITLE
better support for enum type in open api doc

### LINKED ISSUE #7719: Implement enum rendering in `EndpointMetaData`
We currently not support enum drop-down when rendering swagger doc

https://ethereum.github.io/beacon-APIs/#/Beacon/publishBlockV2

image

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
Extend the enum type for the open api doc:
- Add example and description
- Render the enum parameter as a drop down list
- Use the builder pattern to create `EnumTypeDefinition`

## Fixed Issue(s)

fixes #7719 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by mehdi-aouadi** [comment]
Result:
Screenshot 2024-08-23 at 19 23 47

File: infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/EnumTypeDefinition.java [hunk 1]
```diff
@@ -90,6 +120,61 @@ public T deserializeFromString(final String value) {
...
+      return this;
+    }
+
+    public EnumTypeDefinition<T> build() {
+      return new EnumTypeDefinition<>(
+          itemType, serializer, example, description, format, excludedEnumerations);
+    }
+  }
+
   @Override
```
[the remark this answers is not in the document]
Comment (gfukushima) [reply infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/EnumTypeDefinition.java:178]: think this suggestion from greptile makes sense...

**Review by gfukushima — APPROVED** [review]
lgtm

File: infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/EnumTypeDefinition.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mehdi-aouadi) [reply infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/EnumTypeDefinition.java:178]: Definitely. Added the new fields

**Pull request merged**